lib/cli-engine/formatters/text-table.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * @fileoverview Optimized version of the `text-table` npm module to improve performance by replacing inefficient regex-based
+ * whitespace trimming with a modern built-in method.
+ *
+ * @author Nitin Kumar
+ *
+ * This modification addresses a performance issue reported in https://github.com/eslint/eslint/issues/18709
+ *
+ * The `text-table` module is published under the MIT License. For the original source, refer to:
+ * https://www.npmjs.com/package/text-table.
+ */
+
+"use strict";
+
+function dotindex(c) {
+    const m = /\.[^.]*$/u.exec(c);
+
+    return m ? m.index + 1 : c.length;
+}
+
+function reduce(xs, f, init) {
+    if (xs.reduce) {
+        return xs.reduce(f, init);
+    }
+    let i = 0;
+    const acc = arguments.length >= 3 ? init : xs[i++];
+
+    for (; i < xs.length; i++) {
+        f(acc, xs[i], i);
+    }
+    return acc;
+}
+
+function forEach(xs, f) {
+    if (xs.forEach) {
+        return xs.forEach(f);
+    }
+    for (let i = 0; i < xs.length; i++) {
+        f.call(xs, xs[i], i);
+    }
+}
+
+function map(xs, f) {
+    if (xs.map) {
+        return xs.map(f);
+    }
+    const res = [];
+
+    for (let i = 0; i < xs.length; i++) {
+        res.push(f.call(xs, xs[i], i));
+    }
+    return res;
+}
```

Comment:
ESLint doesn't support runtimes that don't have `Array`'s `forEach`, `map`, or `reduce`. These polyfills aren't needed.

```suggestion
```

For reference, my reimplementation that doesn't include any polyfills for old runtimes: 
https://github.com/JoshuaKGoldberg/text-table-fast/blob/a835b26d31ac4645900107ce7640fb950f5b0ac9/src/index.ts

Reply:
:+1:

Reply:
Yeah, makes sense. Removed.